workflow/controller/operator.go

```diff
@@ -901,11 +907,21 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 				woc.updated = true
 			}
 			node := woc.wf.Status.Nodes[pod.ObjectMeta.Name]
+			match := true
+			if woc.execWf.Spec.PodGC != nil && woc.execWf.Spec.PodGC.LabelSelector != nil {
```

Comment:
I sugges you add a `PodGC.GetLabelSelector()`  func that can cope with nil PodGC